master/buildbot/www/service.py

```diff
@@ -153,6 +154,8 @@ def _openLogFile(self, path):
 
     def setSessionSecret(self, secret):
         self.session_secret = secret
+        # change the key for api tokens
+        self.api_token_secret = secret[SESSION_SECRET_LENGTH//8:] + secret[:SESSION_SECRET_LENGTH//8]
```

Comment:
I think it would be better to just hash the secret.

Reply:
You want to use another secret to sign session cookies and api tokens, please use an established salted Key Derivation Function (KDF, e.g. an scrypt pwhash as provided by Libsodium or a simple HMAC HKDF, if there is no other python reference implementation).

By providing a salt (ideally some random bytes, hardcoded as a constant) you can avoid accidentially deriving this same key for another less secure usecase somewhere else.

You also might want to offer adding a secret from the filesystem config file to avoid key leaks when database contents can be seen.